docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
+
+This tutorial demonstrates how to create a brand new Zowe CLI plug-in that uses Node.js to create a client-side API.
+
+After following all the steps, you will have created a data set diff utility plug-in called **Files Util Plug-in**. The plug-in provides an output that can be sent to a third-party utility to display easy-to-read side-by-side diffs of data set member contents, as shown in the image at the [bottom of this page](../extend-cli/cli-developing-a-plugin#bringing-together-new-tools).
+
+The completed source for this tutorial can be found on the `develop-a-plugin` branch of the [zowe-cli-sample-plugin repository](https://github.com/zowe/zowe-cli-sample-plugin).
+
+**Note:** If you are ready to create your own unique Zowe CLI plug-in, refer to the notes below each tutorial step for guidance.
 
-At the end of this tutorial, you will have created a data set diff utility plug-in for Zowe CLI, from which you can pipe
-your plugin's output to a third-party utility for a side-by-side diff of data set member contents.
+## Setting up the new sample plug-in project
+ Download the sample plug-in source and delete the irrelevant content to set up your plug-in project.
+ 
+ Follow these steps:
 
-![Side by Side Diff](../../images/guides/CLI/htmlDiff.png)
+1. Open a terminal and run the command `mkdir zowe-tutorial`.
+    
+    **Note:** All the files created through this tutorial are saved in this tutorial directory.
+2. Enter `cd zowe-tutorial` to change directory into your `zowe-tutorial` folder.
+3. Download the [source code zip file](https://github.com/zowe/zowe-cli-sample-plugin/archive/refs/heads/master.zip) from the Zowe CLI sample plug-in repository.
+4. In your File Explorer, extract the zip file to the  `zowe-tutorial` folder.
+5. Rename the `zowe-cli-sample-plugin-master` directory to `files-util`.
+    
+    This is the project directory used throughout the rest of this tutorial. 
+6. Delete all content within the following folders:
+    - `src/api`
+    - `src/cli`
+    - `docs` folders
+    - `__tests__/__system__/api`
+    - `__tests__/__system__/cli`
+    -  `__tests__/api`
+    - `__tests__/cli`
+7. Return to your terminal and run `cd files-util` to enter the project directory.
+8. Enter `git init` to set up a new Git repository.
+9. Enter `git add --all` to *stage* (track) all files in the current directory with Git.
 
-Completed source for this tutorial can be found on the `develop-a-plugin` branch of the zowe-cli-sample-plugin repository.
+10. Enter `git commit --message "Initial commit"` to save a snapshot of the staged files in your repository.
+11. Run `npm install` to install third-party dependencies defined in the `package.json` file of your Node.js project.
 
-### Cloning the sample plug-in source
- Clone the sample repo, delete the irrelevant source, and create a brand new plug-in. Follow these steps:
+    When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
```

Comment:
There will be a progress bar...
- ![image](https://user-images.githubusercontent.com/37381190/219688417-64e399bc-867e-4b13-9208-11557d3dbac5.png)
- ![image](https://user-images.githubusercontent.com/37381190/219688474-78d4410c-44c4-49ef-875d-2daf186690be.png)

And finally the results of the installation.
- ![image](https://user-images.githubusercontent.com/37381190/219688602-7d4b79f1-e794-4872-8d62-36c9f09fd3dd.png)

Reply:
Since the package may have vulnerabilities, should we document how to fix them? Or maybe just point them to NPM's doc regarding the `npm audit` command? Or simply say "follow the prompts"? 😋